### ISSUE TITLE
false positive W0718

### ISSUE THREAD

**Issue by bb-ben** [pylint-dev/pylint#8308]
### Bug description

Looks like W0718 will not be reported when Exception is used in the `except` clause. For example

```python
# main.py
try:
    with open("does_not_exists", "r") as f:
        f.read()
except Exception as e:
    raise e  # with `e` being used here, W0718 will be suppressed
```

### Command used

```shell
pylint main.py
```

### Pylint output

```shell
************* Module main
main.py:1:0: C0114: Missing module docstring (missing-module-docstring)
main.py:2:9: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)

------------------------------------------------------------------
Your code has been rated at 6.00/10 (previous run: 6.00/10, +0.00)
```

### Expected behavior

In this case, W0718(broad-exception-caught) should be in the report, and perhaps `raise e` should result in W0719(broad-exception-raised) too.

### Pylint version

```shell
pylint 2.16.2
astroid 2.14.2
Python 3.10.9 (main, Feb 13 2023, 10:15:31) [GCC 9.4.0]
```

### OS / Environment

Ubuntu 20.04 LTS (Focal Fossa)

**Comment by jacobtylerwalls** [comment]
Thanks for the report, but this is working as designed. Fiddling with the error message and then immediately re-raising is functionally equivalent to not catching at all, which is the wanted pattern.

See:
https://github.com/pylint-dev/pylint/blob/c51f18f2ac6bedc7574938081e55a54da0ec845f/pylint/checkers/exceptions.py#L569
